pkg/splunk/enterprise/clustermaster.go

```diff
@@ -119,6 +123,10 @@ func ApplyClusterMaster(client splcommon.ControllerClient, cr *enterprisev1.Clus
 
 	// no need to requeue if everything is ready
 	if cr.Status.Phase == splcommon.PhaseReady {
+		err = ApplyMonitoringConsole(client, cr, cr.Spec.CommonSplunkSpec, getClusterMasterExtraEnv(cr, &cr.Spec.CommonSplunkSpec))
+		if err != nil {
+			return result, err
+		}
```

Comment:
Though it is not harmful, Isn't it expensive to call again and again? Is it a good idea to detect that there is a deployment already in place, and that way bailout here, instead of going deep inside? 

Also, can you please confirm, if all those REST calls are going to be executed by any chance in this path?

Reply:
My questions is a bit different, let us discuss